src/adagrams.js

```diff
@@ -1,15 +1,141 @@
+
+// letter frequency  Object notation to model a Dictionary
+const POOL_OF_LETTERS = {
+  'A': 9, 'B': 2, 'C': 2, 'D': 4, 'E': 12,
+  'F': 2, 'G': 3, 'H': 2, 'I': 9, 'J': 1,
+  'K': 1, 'L': 4, 'M': 2, 'N': 6, 'O': 8,
+  'P': 2, 'Q': 1, 'R': 6, 'S': 4, 'T': 6,
+  'U': 4, 'V': 2, 'W': 2, 'X': 1, 'Y': 2, 'Z': 1
+};
+
+export const computeCumulativeWeights = (letterPool) => {
+  const cumulativeWeightsDict = {};
+  let runningSum = 0;
+
+  for (const letter in letterPool) {
+    runningSum += letterPool[letter];
+    cumulativeWeightsDict[letter] = runningSum;
+  }
+  
+  return [cumulativeWeightsDict, runningSum];
+};
+
+
+export const drawRandomNums = (maxNum) => {
+  const randomNums = new Set();
+
+  while (randomNums.size !== 10){
+    const randomNum = Math.floor(Math.random() * maxNum);
+    randomNums.add(randomNum)
+  };
+  return Array.from(randomNums);
+};
+
+// Main Logic – Draw Letters
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  // map numbers to specific letters
+  const result = computeCumulativeWeights(POOL_OF_LETTERS);
+  const cumulativeWeightsDict = result[0];
+  const totalWeight = result[1];
```

Comment:
A couple of things here: 
- The variable `result` is not really descriptive of the data that it is holding. 
- To avoid the `result` variable, you could using destructuring like how we do with state in React to save the two elements of the returned array to their own variables like so: 
```suggestion
  const [cumulativeWeightsDict, totalWeight] = computeCumulativeWeights(POOL_OF_LETTERS);
```